test/python/quantum_info/test_synthesis.py

```diff
@@ -427,148 +443,135 @@ def test_cnot_rxx_decompose(self):
             self.assertTrue(cnot.equiv(decomp))
 
-    def test_exact_two_qubit_cnot_decompose_random(self, nsamples=10):
-        """Verify exact CNOT decomposition for random Haar 4x4 unitaries.
+    @combine(seed=range(10), name='test_exact_two_qubit_cnot_decompose_random_{seed}')
+    def test_exact_two_qubit_cnot_decompose_random(self, seed):
+        """Verify exact CNOT decomposition for random Haar 4x4 unitary (seed={seed}).
         """
-        for _ in range(nsamples):
-            unitary = random_unitary(4)
-            self.check_exact_decomposition(unitary.data, two_qubit_cnot_decompose)
+        unitary = random_unitary(4, seed=seed)
+        self.check_exact_decomposition(unitary.data, two_qubit_cnot_decompose)
 
     def test_exact_two_qubit_cnot_decompose_paulis(self):
         """Verify exact CNOT decomposition for Paulis
         """
         unitary = Operator.from_label('XZ')
         self.check_exact_decomposition(unitary.data, two_qubit_cnot_decompose)
 
-    def test_exact_supercontrolled_decompose_random(self, nsamples=10):
-        """Verify exact decomposition for random supercontrolled basis and random target"""
-
-        for _ in range(nsamples):
-            k1 = np.kron(random_unitary(2).data, random_unitary(2).data)
-            k2 = np.kron(random_unitary(2).data, random_unitary(2).data)
-            basis_unitary = k1 @ Ud(np.pi/4, 0, 0) @ k2
-            decomposer = TwoQubitBasisDecomposer(UnitaryGate(basis_unitary))
-            self.check_exact_decomposition(random_unitary(4).data, decomposer)
+    @combine(seed=range(10), name='test_exact_supercontrolled_decompose_random_{seed}')
+    def test_exact_supercontrolled_decompose_random(self, seed):
+        """Exact decomposition for random supercontrolled basis and random target (seed={seed})"""
+        # pylint: disable=invalid-name
+        k1 = np.kron(random_unitary(2, seed=seed).data, random_unitary(2, seed=seed + 1).data)
+        k2 = np.kron(random_unitary(2, seed=seed + 2).data, random_unitary(2, seed=seed + 3).data)
+        basis_unitary = k1 @ Ud(np.pi / 4, 0, 0) @ k2
+        decomposer = TwoQubitBasisDecomposer(UnitaryGate(basis_unitary))
+        self.check_exact_decomposition(random_unitary(4, seed=seed + 4).data, decomposer)
 
     def test_exact_nonsupercontrolled_decompose(self):
         """Check that the nonsupercontrolled basis throws a warning"""
         with self.assertWarns(UserWarning, msg="Supposed to warn when basis non-supercontrolled"):
-            TwoQubitBasisDecomposer(UnitaryGate(Ud(np.pi/4, 0.2, 0.1)))
+            TwoQubitBasisDecomposer(UnitaryGate(Ud(np.pi / 4, 0.2, 0.1)))
 
-    def test_cx_equivalence_0cx_random(self):
-        """Check random circuits with  0 cx
-        gates locally equivalent to identity
+    def test_cx_equivalence_0cx(self, seed=0):
+        """Check circuits with  0 cx gates locally equivalent to identity
         """
+        state = np.random.RandomState(seed)
```

Comment:
`RandomState` was recently changed to `default_rng` - this change needs to be applied throughout.

```suggestion
        state = np.random.default_rng(seed)
```

Reply:
added! good catch!

Reply:
there are still leftovers

Reply:
done in d09b04063